src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions_Negate.cs

```diff
@@ -44,37 +48,71 @@ internal static partial class SyntaxGeneratorExtensions
                         cancellationToken))
-                    .WithTriviaFrom(expression);
-            }
-            if (negateBinary && syntaxFacts.IsBinaryExpression(expression))
-            {
-                return GetNegationOfBinaryExpression(expression, generator, generatorInternal, semanticModel, cancellationToken);
-            }
-            else if (syntaxFacts.IsLiteralExpression(expression))
-            {
-                return GetNegationOfLiteralExpression(expression, generator, semanticModel);
+                    .WithTriviaFrom(expressionOrPattern);
             }
-            else if (syntaxFacts.IsLogicalNotExpression(expression))
+
+            if (negateBinary && syntaxFacts.IsBinaryExpression(expressionOrPattern))
+                return GetNegationOfBinaryExpression(expressionOrPattern, generator, generatorInternal, semanticModel, cancellationToken);
+
+            if (syntaxFacts.IsLiteralExpression(expressionOrPattern))
+                return GetNegationOfLiteralExpression(expressionOrPattern, generator, semanticModel);
+
+            if (syntaxFacts.IsLogicalNotExpression(expressionOrPattern))
+                return GetNegationOfLogicalNotExpression(expressionOrPattern, syntaxFacts);
+
+#if CODE_STYLE
+            return generator.LogicalNotExpression(expressionOrPattern);
+#else
+            if (negateBinary && syntaxFacts.IsIsPatternExpression(expressionOrPattern))
+                return GetNegationOfIsPatternExpression(expressionOrPattern, generator, generatorInternal, semanticModel, cancellationToken);
+
+            if (syntaxFacts.IsParenthesizedPattern(expressionOrPattern))
             {
-                return GetNegationOfLogicalNotExpression(expression, syntaxFacts);
+                // Push the negation inside the parenthesized pattern.
+                return generatorInternal.AddParentheses(
+                    generator.Negate(
+                        generatorInternal,
+                        syntaxFacts.GetPatternOfParenthesizedPattern(expressionOrPattern),
+                        semanticModel,
+                        negateBinary,
+                        cancellationToken))
+                    .WithTriviaFrom(expressionOrPattern);
             }
 
-            return generator.LogicalNotExpression(expression);
+            if (negateBinary && syntaxFacts.IsBinaryPattern(expressionOrPattern))
+                return GetNegationOfBinaryPattern(expressionOrPattern, generator, generatorInternal, semanticModel, cancellationToken);
+
+            if (syntaxFacts.IsConstantPattern(expressionOrPattern))
+                return GetNegationOfConstantPattern(expressionOrPattern, generator, generatorInternal);
+
+            if (syntaxFacts.IsUnaryPattern(expressionOrPattern))
+                return GetNegationOfUnaryPattern(expressionOrPattern, generator, syntaxFacts);
+
+            // TODO(cyrusn): We could support negating relational patterns in the future.  i.e.
```

Comment:
Tracking issue instead of (cyrusn)?

Reply:
i don't find tracking issue useful as they don't get looked at, and eventually bulk closed.